cli/cmd/top.go

```diff
@@ -348,30 +348,44 @@ func renderHeaders(withSource bool) {
 	}
 }
 
+func max(i, j int) int {
+	if i > j {
+		return i
+	}
+	return j
+}
+
 func renderTableBody(table *[]tableRow, withSource bool) {
 	sort.SliceStable(*table, func(i, j int) bool {
 		return (*table)[i].count > (*table)[j].count
 	})
+	adjustedColumnWidths := columnWidths
+	for _, row := range *table {
+		adjustedColumnWidths[0] = max(adjustedColumnWidths[0], runewidth.StringWidth(row.source))
+		adjustedColumnWidths[1] = max(adjustedColumnWidths[1], runewidth.StringWidth(row.destination))
+		adjustedColumnWidths[3] = max(adjustedColumnWidths[3], runewidth.StringWidth(row.by))
+
+	}
 	for i, row := range *table {
 		x := 0
 		if withSource {
-			tbprint(x, i+headerHeight, row.source)
-			x += columnWidths[0] + 1
+			tbprint(x, i+headerHeight, runewidth.Truncate(row.source, adjustedColumnWidths[0], ""))
```

Comment:
If we are adjusting the column width to the maximum runewidth in that column, is this truncation needed?

Reply:
Oops you're right. That was a leftover from a previous iteration. I've just removed it.